pkg/p2p/libp2p/libp2p.go

```diff
@@ -278,7 +279,7 @@ func (s *Service) AddProtocol(p p2p.ProtocolSpec) (err error) {
 			s.metrics.HandledStreamCount.Inc()
 			if err := ss.Handler(ctx, p2p.Peer{Address: overlay}, stream); err != nil {
 				var e *p2p.DisconnectError
-				if errors.Is(err, e) {
+				if errors.As(err, &e) {
```

Comment:
This is a bug fix, not related to the scope of this PR really, but I saw it while debugging.

Reply:
Cool, but I think that if `e` is already a pointer, it is enough to call `errors.As(err,e)`, you're passing a pointer to a pointer, I'm not sure this is necessary. Another thing is that `e` is never used. Why not just check `Is` with the type directly?

Reply:
Resolved on mattermost 👍